### DISCUSSION TITLE
Could you change that timelineVariable return an error when an improper argument is specified?

### DISCUSSION THREAD

**Discussion by kurokida** [jspsych/jsPsych#2679] [Ideas]
Thanks to everyone involved in development of jsPsych!

I'm developing [a psychophysics plugin](https://jspsychophysics.hes.kyushu-u.ac.jp/).

One of the users of my plugin had set a property as follows:

```javascript
response_start_time: jsPsych.timelineVariable('circle_start_time'),
```

But he forgot to define the circle_start_time.

In this case, the timelineVariable function did not return an error and the response_start_time property was set to its default value of 0.

I think this behavior makes the bug harder to notice and concerns the use of jsPsych beyond my plugin.
If possible, the timelineVariable function should return an error when a value which is not in timeline_variables is specified.

Best regards,
Daiichiro

**Comment by jodeleeuw** [comment]
Thanks for flagging this. This makes sense to me. I'll work on a patch.
